### ISSUE TITLE
a11y: Element with role combobox missing accessible name

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/auro-combobox#219]
### Please verify the version of auro-combobox you have installed

@latest

### Please describe the bug

See the following code:

https://github.com/AlaskaAirlines/auro-combobox/blob/6e23ce936eae93d2dcc40adf00d236972b940a5d/src/auro-combobox.js#L712-L758

One or more elements that has a role='combobox' attribute, not intentionally hidden in the DOM and available to assistive technologies, does not have a mechanism that allows an accessible name to be calculated.

### Reproducing the error

This issue is reproducible on the Auro docsite

### Expected behavior

When elements with `role` of `combobox` do not have an accessible name, assistive technologies may incorrectly guess at a name or provide no name at all to users. When an accessible name is not present, users of assistive technologies may not be able to complete a form.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Additional context

The following code changes are recommended to simplify the DOM rendering of the element and increase accessibility awareness for screen reader users. 

1. remove the `for` attribute. Does not appear to be used anywhere else in the DOM
2. add `aria-labelledby="combobox-label"` for accessible name association

```html
-          for="dropdownMenu"
+          aria-labelledby="combobox-label"
           bordered
```

1. Add `id="combobox-label"`

```html
-            <slot name="label" slot="label"></slot>
+            <slot name="label" slot="label" id="combobox-label"></slot>
```

**Additional recommendation**

Within the `render()` function the return HTML is wrapped in an undefined `div` element. This is not required for LitElement and serves no real purpose in the template. Recommend removing wrapping `div` element.

**Additional state management** 

The following aria roles need to be considered, but undetermined where they should be added within this context. The issue of the `role="button"` currently makes this difficult to determine. a11yUX needs to be reviewed once issue https://github.com/AlaskaAirlines/auro-dropdown/issues/273 is addressed. 

```html
aria-expanded="false"
aria-haspopup="listbox"
aria-owns="listbox"
aria-autocomplete="list"
aria-activedescendant="" // takes as its value the id of the currently focused element.
aria-controls="listbox"
```

**Comment by blackfalcon** [comment]
The severity of the issue was updated to CRITICAL due to an active complaint from the DOT regarding accessibility in UIs supported by this element. 

See ADO story for additional details https://itsals.visualstudio.com/E_Sell_Shop/_workitems/edit/1144649

**Comment by blackfalcon** [comment]
Related issue

- https://github.com/AlaskaAirlines/auro-dropdown/issues/273
